### PR TITLE
Fix #780, add spaces in parens flag

### LINKED ISSUE #780: bring back spaces in parenthesis
Could we consider bringing back spaces in parentheses :)? https://github.com/olafurpg/scalafmt/issues/116

I personally find it much easier to parse (i.e. read) code with spaces inside of parenthesis. To the point that this is a necessary requirement for me to be able to use scalafmt. Other people at @xdotai like this style as well and we use it across our code base. I know we seem to be in the minority about this aspect, but we'd really love to migrate from scalariform to scalafmt and this is blocking us right now :).

### PR DESCRIPTION
**Pull request by olafurpg**
cc/ @cvogt

### CODE REVIEW

File: core/src/test/resources/spaces/InParentheses.stat [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+function()
+<<< def
+def x(foo: Int) = 2
+>>>
+def x( foo: Int ) = 2
+<<< tuple
+def x = (1, 2)
+>>>
+def x = ( 1, 2 )
+<<< maxColumn (looks weird, but scalariform does the same)
```
Comment (olafurpg) [inline core/src/test/resources/spaces/InParentheses.stat]: What do you think about these @cvogt?

Comment (cvogt) [inline core/src/test/resources/spaces/InParentheses.stat]: Interesting. Never seen that with scalariform.

Is it hard to let it produce this instead?
```
def x =
  ( 1,
    2,
    aaaaaaaaaaaaaawwwwsssssssssswa )
```

Are there any other flags I can set to let it produce either of:
```
def x = (
  1,
  2,
  aaaaaaaaaaaaaawwwwsssssssssswa )
```
```
def x = (
  1,
  2,
  aaaaaaaaaaaaaawwwwsssssssssswa
)
```

```
def x = (
  1, 2,
  aaaaaaaaaaaaaawwwwsssssssssswa
)
```
?

**Comment by cvogt** [comment]
@olafurpg
I personally don't use spaces.inBrackets at the moment.

I believe some people like to distinguish call sites from definition sites for whitespace in parenthesis, but I personally like to have it in both, so the current solution works for me in this regard.

File: core/src/test/resources/spaces/InParentheses.stat (hunk 1 above)
Comment (olafurpg) [inline core/src/test/resources/spaces/InParentheses.stat]: Seems like you want `style = IntelliJ`, that will produce these outputs

```scala
[info]   ===========
[info]   => Obtained
[info]   ===========
[info]   def foo =
[info]     bar(
[info]       aaaaaaaaaaaaa,
[info]       bbbbbbbbbbbbbb,
[info]       cccccccccc
[info]     )
[info]   ===========
[info]   => Obtained
[info]   ===========
[info]   def x =
[info]     (
[info]       1,
[info]       2,
[info]       aaaaaaaaaaaaaawwwwsssssssssswa
[info]     )

```

**Comment by olafurpg** [comment]
I'll go ahead and merge this since it seems `style = IntelliJ` will handle the weird max column tests. You are the only user of this new option, so I won't bother with `inBrackets` or distinguishing def/call site

**Comment by cvogt** [comment]
thx :D. I will check it out.

**Pull request merged**